src/ProjectTracker1L.java

```diff
@@ -0,0 +1,110 @@
+package components.projecttracker;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Kernel implementation for ProjectTracker.
+ */
+public class ProjectTracker1L extends ProjectTrackerSecondary {
+
+    /**
+     * Representation of a milestone.
+     */
+    private static class Milestone {
```

Comment:
If this is read-only, you might consider using a Java record. I suspect it's not because of the status variable.